### PR TITLE
BYOC

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/3316

Implementation of the BYOC feature on azure-operator

It includes:
* reading guest cluster credential on runtime
* guest cluster scoped azure client
* replacement of vnet peering with vpn gateway

### CODE REVIEW

File: integration/template/azure_operator_charts_values.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ var AzureOperatorChartValues = `Installation:
           CIDR: "10.0.0.0/16"
           ResourceGroup: "godsmack"
           VirtualNetwork: "godsmack"
+          VirtualNetworkGateway: "GatewayTest"
```
Comment (TheoBrigitte) [inline integration/template/azure_operator_charts_values.go]: TODO: align with godsmack vpn gateway

**Comment by xh3b4sd** [comment]
What do you think about providing different PRs so we get changes out gradually. This here is a bit big. Is there a reason why we have to introduce all of these changes at once?

**Comment by TheoBrigitte** [comment]
Moving from regular azure client to guest cluster scoped client, makes it hard to decouple some parts. But I can provide PR for other changes.

**Comment by xh3b4sd** [comment]
I claim you can provide a PR for each bullet point in your PR description. 10 XRP say you can.

**Comment by TheoBrigitte** [comment]
Closing, as this PR has been split

https://github.com/giantswarm/azure-operator/pull/263
https://github.com/giantswarm/azure-operator/pull/264
https://github.com/giantswarm/azure-operator/pull/265
https://github.com/giantswarm/azure-operator/pull/266
https://github.com/giantswarm/azure-operator/pull/267

**Pull request closed without merging**